scripts/mk_dec_eq.hs

```diff
@@ -0,0 +1,62 @@
+
```

Comment:
I'd like to see more documentation in general (at least document the purpose of each file), but some documentation on this file in particular would be really nice.  This may be broadly useful, but I have to dig into the code to figure out what it does.